mlir/include/mlir/Dialect/Presburger/Parser.h

```diff
@@ -260,12 +267,51 @@ class SetExpr : public Expr {
   std::unique_ptr<Expr> constraints;
 };
 
+class PieceExpr : public Expr {
+public:
+  PieceExpr(std::unique_ptr<Expr> expr, std::unique_ptr<Expr> constraints)
+      : expr(std::move(expr)), constraints(std::move(constraints)) {}
+
+  Expr *getExpr() { return expr.get(); }
+  Expr *getConstraints() { return constraints.get(); }
+
+  static Type getStaticType() { return Type::Piece; }
+  virtual Type getType() { return Type::Piece; }
+
+private:
+  std::unique_ptr<Expr> expr;
+  std::unique_ptr<Expr> constraints;
+};
+
+class PwExprExpr : public Expr {
```

Comment:
I'm not really happy with the structure of this class, but I didn't find a nicer way yet.